### PR TITLE
use a __version__ variable, update setup.py to use it

### PR DESCRIPTION
**Pull request by chmreid**
This borrows/utilizes some code from [black](https://github.com/psf/black) to allow macchiato to define a `__version__` variable in `macchiato.py` and use that to set the version number in `setup.py`. This also allows the user to programmatically obtain the version of `macchiato`, which is currently not possible to do, from either the Python API or the command line.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,4 +1,19 @@
 from setuptools import setup
+from pathlib import Path
+import re, ast
```
Comment (wbolster) [inline setup.py]: please split over multiple lines, and sort the imports

**Review by wbolster — CHANGES_REQUESTED** [review]
some nitpicks but fine overall. thanks

File: setup.py (hunk 1 above)
Comment (chmreid) [reply setup.py]: changes have been made, thanks for reviewing!

**Review by wbolster — APPROVED** [review]

**Comment by wbolster** [comment]
thanks!

**Pull request merged**